src/ConfigHandler.h

```diff
@@ -18,8 +20,9 @@ class ConfigHandler {
   ~ConfigHandler() = default;
   static std::shared_ptr<ConfigHandler> const getInstance();
 
-  void setConfigFilePath(std::string filepath);
-  TResult<std::string> getValue(std::string section, std::string key);
+  TResultOpt setConfigFilePath(std::string filepath);
+  TResult<std::string> getValueString(std::string section, std::string key);
```

Comment:
if you only read your string parameters, use `const&` to avoid copying

Reply:
applies for all parameters which are greater than a pointer or to parameter types which do some heap allocation or that kind of stuff